### ISSUE TITLE
VB.NET - Else with Expression doesn't throw Compiler Error

### ISSUE THREAD

**Issue by medoni** [dotnet/roslyn#47109]
The following code doesn't throw a compiler error.
```VB.NET
Dim str = "Foo"
If str.StartsWith("bar") Then
    Console.WriteLine("bar")
Else str.StartsWith("baz") 
    Console.WriteLine("baz")
End If
```
Hint, the `Else` should be an `Else If`

**Version Used**: 
Microsoft Visual Studio Enterprise 2019
Version 16.7.1

**Expected Behavior**:
A compile error (missing if keyword) at the `Else` line

**Actual Behavior**:
No error. The output on console is *baz*

**Comment by Youssef1313** [comment]
This absolutely looks like a bug. I'll give this a try soon.

**Comment by RikkiGibson** [comment]
This also happens in VS 2017 (using Roslyn 2.10.x).

That makes it seem like we've had this bug for a long time, so I am wondering whether it is an acceptable breaking change to fix it. /cc @jaredpar

**Comment by RikkiGibson** [comment]
Also wondering if this is a misunderstanding and the language simply allows a statement on the same line as the `Else`, but doesn't use it as a condition.

**Comment by Youssef1313** [comment]
@RikkiGibson The native compiler produces an error for it.

**Comment by PathogenDavid** [comment]
> Also wondering if this is a misunderstanding and the language simply allows a statement on the same line as the `Else`, but doesn't use it as a condition.

I don't believe so. Looking at the spec, it doesn't seem like you should be able to have anything after `Else`:

The syntax of [`Else` is defined as](https://github.com/dotnet/vblang/blob/19ec79a287fb79180b05a0ad20e8291e75fc63df/spec/statements.md#ifthenelse-statements):

```antlr
ElseStatement
    : 'Else' StatementTerminator
      Block?
    ;
```

Where [`StatementTerminator` is defined as](https://github.com/dotnet/vblang/blob/19ec79a287fb79180b05a0ad20e8291e75fc63df/spec/source-files-and-namespaces.md):

```antlr
StatementTerminator
    : LineTerminator
    | ':'
    ;
```

And [`LineTerminator` is defined as](https://github.com/dotnet/vblang/blob/19ec79a287fb79180b05a0ad20e8291e75fc63df/spec/lexical-grammar.md#line-terminators):

```antlr
LineTerminator
    : '<Unicode 0x00D>'
    | '<Unicode 0x00A>'
    | '<CR>'
    | '<LF>'
    | '<Unicode 0x2028>'
    | '<Unicode 0x2029>'
    ;
```

-------------

In the examples given, there's no `StatementTerminator` between the `Else` and the method invocation.

However as [I noted here](https://github.com/dotnet/roslyn/pull/47112#issuecomment-693332947), this bug seems to be as old as time and the online compiler website Youssef used is actually using the Mono compiler (which seems to be correct as far as the spec is concerned.)

**Comment by AlekseyTs** [comment]
It looks like the behavior goes back to the native VB compiler and the code that follows the ```Else``` is getting executed as if it was in the ```Else``` block.

**Comment by Youssef1313** [comment]
@AlekseyTs So will it be just left as-is? or will it be fixed (either by error or a warning in a warning wave)? or will the spec be updated to reflect this?

**Comment by AlekseyTs** [comment]
We decided to leave the behavior as is, i.e. we acknowledge that this is a bug, but will not be fixing it. We will not be adjusting specification as well.
